From e9df0c725625dc097a5766b259bf354ee01f7bd1 Mon Sep 17 00:00:00 2001
From: He Sheng <hesheng@wxiat.com>
Date: Thu, 13 Oct 2022 11:01:10 +0800
Subject: [PATCH] sw64: fix compile error for CONFIG_RRU=y
Patch-mainline: Not yet, from openEuler
References: bsn#22
openEuler-commit: e9df0c725625dc097a5766b259bf354ee01f7bd1
Modified-by-SEL: No


Sunway inclusion
category: feature
bugzilla: https://gitee.com/openeuler/kernel/issues/I56QAM

--------------------------------

Signed-off-by: He Sheng <hesheng@wxiat.com>
Reviewed-by: Cui Wei <cuiwei@wxiat.com>
Signed-off-by: Gu Zitao <guzitao@wxiat.com>
Signed-off-by: Guoqing Jiang <guoqing.jiang@suse.com>
---
 arch/sw_64/kernel/dup_print.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/arch/sw_64/kernel/dup_print.c b/arch/sw_64/kernel/dup_print.c
index 02639f40a4bc..3a32c444207d 100644
--- a/arch/sw_64/kernel/dup_print.c
+++ b/arch/sw_64/kernel/dup_print.c
@@ -52,6 +52,8 @@ int sw64_printk(const char *fmt, va_list args)
 #endif
 
 #ifdef CONFIG_SW64_RRU
+#include<linux/uaccess.h>
+
 static DEFINE_SPINLOCK(printf_lock);
 #define USER_PRINT_BUFF_BASE            (0x600000UL + __START_KERNEL_map)
 #define USER_PRINT_BUFF_LEN             0x100000UL
-- 
2.33.0

